tests/config_snmp_test.py

```diff
@@ -0,0 +1,814 @@
+        print(result.exit_code)
+        assert result.exit_code == 2
+        assert 'FAILED: SNMP community string length should be not be greater than 32' in result.output
+
+
+    # Del snmp contact when CONTACT not setup in REDIS
+    def test_config_snmp_contact_del_without_contact_redis(self):
+        db = Db()
+        runner = CliRunner()
+        result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["del"], ["blah"], obj=db)
+        print(result.exit_code)
+        assert result.exit_code == 2
+        assert 'Contact name blah is not configured' in result.output
+        assert db.cfgdb.get_entry("SNMP", "CONTACT") == {}
+
+    def test_config_snmp_contact_modify_without_contact_redis(self):
+        db = Db()
+        runner = CliRunner()
+        result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["modify"], 
+                                                     ["blah", "blah@contoso.com"], obj=db)
+        print(result.exit_code)
+        assert result.exit_code == 2
+        assert 'Contact name blah is not configured' in result.output
+        assert db.cfgdb.get_entry("SNMP", "CONTACT") == {}
+
+    def test_config_snmp_contact_add_del_new_contact(self):
+        db = Db()
+        runner = CliRunner()
+        with mock.patch('utilities_common.cli.run_command') as mock_run_command:
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                    ["testuser", "testuser@contoso.com"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert result.output == config_snmp_contact_add_del_new_contact
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {"testuser": "testuser@contoso.com"}
+
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["del"],
+                                    ["testuser"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert 'SNMP contact testuser removed from configuration' in result.output
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {} 
+
+    # Add snmp contact tests
+    def test_config_snmp_contact_add_with_existing_contact(self):
+        db = Db()
+        runner = CliRunner()
+        with mock.patch('utilities_common.cli.run_command') as mock_run_command:
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                    ["testuser", "testuser@contoso.com"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert result.output == config_snmp_contact_add_del_new_contact
```

Comment:
So this is testing if there is an existing contact and what happens if we try to add a 2nd one.  The point is to not allow the 2nd one to be added. 
I have to add the original testuser contact first since we have no contact in the mock_table/config_db.json

Reply:
Got the idea behind adding a contact first. The second contact cannot be added while first exists.

Reply:
Yep.   Thanks Suvarna.